code/game/objects/items/RSF.dm

```diff
@@ -8,178 +8,200 @@ RSF
+		//We then add their data into the radial menu
+		radial_list[test.name] = image(icon = test.icon, icon_state = test.icon_state)
+		qdel(test)
+	return radial_list
+
+/obj/item/rsf/proc/check_menu(mob/user)
+	if(user.incapacitated() || !user.Adjacent(src))
+		return FALSE
+	return TRUE
+
 /obj/item/rsf/afterattack(atom/A, mob/user, proximity)
 	. = ..()
 	if(!proximity)
 		return
-	if (!(istype(A, /obj/structure/table) || isfloorturf(A)))
+	if(!is_allowed(A))
 		return
+	if(use_matter(dispense_cost, user))//If we can charge that amount of charge, we do so and return true
+		playsound(loc, 'sound/machines/click.ogg', 10, TRUE)
+		var/atom/meme = new to_dispense(get_turf(A))
+		to_chat(user, "<span class='notice'>[action_type] [meme.name]...</span>")
 
-	if(iscyborg(user))
-		var/mob/living/silicon/robot/R = user
-		if(!R.cell || R.cell.charge < 200)
-			to_chat(user, "<span class='warning'>You do not have enough power to use [src].</span>")
-			icon_state = "rsf_empty"
-			return
-	else if (matter < 1)
-		to_chat(user, "<span class='warning'>\The [src] doesn't have enough matter left.</span>")
-		icon_state = "rsf_empty"
-		return
-
-	var/turf/T = get_turf(A)
-	playsound(src.loc, 'sound/machines/click.ogg', 10, TRUE)
-	switch(mode)
-		if(1)
-			to_chat(user, "<span class='notice'>Dispensing Drinking Glass...</span>")
-			new /obj/item/reagent_containers/food/drinks/drinkingglass(T)
-			use_matter(20, user)
-		if(2)
-			to_chat(user, "<span class='notice'>Dispensing Paper Sheet...</span>")
-			new /obj/item/paper(T)
-			use_matter(10, user)
-		if(3)
-			to_chat(user, "<span class='notice'>Dispensing Pen...</span>")
-			new /obj/item/pen(T)
-			use_matter(50, user)
-		if(4)
-			to_chat(user, "<span class='notice'>Dispensing Dice Pack...</span>")
-			new /obj/item/storage/pill_bottle/dice(T)
-			use_matter(200, user)
-		if(5)
-			to_chat(user, "<span class='notice'>Dispensing Cigarette...</span>")
-			new /obj/item/clothing/mask/cigarette(T)
-			use_matter(10, user)
-
+///A helper proc. checks to see if we can afford the amount of charge that is passed, and if we can docs the charge from our base, and returns TRUE. If we can't we return FALSE
 /obj/item/rsf/proc/use_matter(charge, mob/user)
-	if (iscyborg(user))
+	if(iscyborg(user))
 		var/mob/living/silicon/robot/R = user
-		R.cell.charge -= charge
+		var/end_charge = R.cell.charge - charge
+		if(end_charge >= 0)
+			R.cell.charge = end_charge
+			return TRUE
+		to_chat(user, "<span class='warning'>You do not have enough power to use [src].</span>")
 	else
-		matter--
-		to_chat(user, "<span class='notice'>The RSF now holds [matter]/30 fabrication-units.</span>")
-
-/obj/item/cookiesynth
+		if(matter - 1 >= 0)
+			matter--
+			to_chat(user, "<span class='notice'>\The [src] now holds [matter]/[max_matter] [discriptor].</span>")
+			return TRUE
+		to_chat(user, "<span class='warning'>\The [src] doesn't have enough [discriptor] left.</span>")
+	icon_state = spent_icon_state
+	return FALSE
```

Comment:
Do early returns, even if it results in a bit more code.
Things like
``cpp
if(iscyborg(user))
	var/mob/living/silicon/robot/R = user
	var/end_charge = R.cell.charge - charge
	if(end_charge < 0)
		to_chat(user, "<span class='warning'>You do not have enough power to use [src].</span>")
		return FALSE
	R.cell.charge = end_charge
	return TRUE
if(matter - 1 < 0)
	to_chat(user, "<span class='warning'>\The [src] doesn't have enough [discriptor] left.</span>")
	icon_state = spent_icon_state
	return FALSE
...
```

Reply:
If you say so. I feel it hurts the readability somewhat, but I don't know a ton about that.